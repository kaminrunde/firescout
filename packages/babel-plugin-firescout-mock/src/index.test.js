var {transformSync} = require('@babel/core')
var plugin = require('.')

const transform = code => transformSync(code, {
  presets: ["@babel/preset-react"],
  plugins: [plugin],
}).code

test('transforms declared arrow functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
const xy = test => { return test }`)
  
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function fn(test) {
  return test;
});`
  )
});

test('transforms declared exported arrow functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
export const xy = test => { return test }`)
  
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
export const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function fn(test) {
  return test;
});`
  )
});

test('transforms default exported arrow functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
export default test => { return test }`)
  
  expect(output).toEqual(
`export default require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function fn(test) {
  return test;
});`
  )
});

test('transforms declared async arrow functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
const xy = async test => { return test }`)
  
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", async function fn(test) {
  return test;
});`
  )
});

test('transforms functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
function xy (test) { return test }`)
  expect(output).toEqual(
`var xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function xy(test) {
  return test;
});`)
});

test('transforms exported functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
export function xy (test) { return test }`)
  expect(output).toEqual(
`export var xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function xy(test) {
  return test;
});`)
});

test('transforms default exported functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
export default function xy (test) { return test }`)
  expect(output).toEqual(
`export default require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function xy(test) {
  return test;
});`)
});

test('transforms default exported anonym functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
export default function (test) { return test }`)
  expect(output).toEqual(
`export default require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function fn(test) {
  return test;
});`)
});

test('transforms async functions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
async function xy (test) { return test }`)
  expect(output).toEqual(
`var xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", async function xy(test) {
  return test;
});`)
});

test('transforms function expressions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
const xy = function xy (test) { return test }`)
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function xy(test) {
  return test;
});`)
});

test('transforms function expressions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
const xy = function (test) { return test }`)
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function fn(test) {
  return test;
});`)
});

test('transforms named function expressions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
const xy = function xy (test) { return test }`)
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function xy(test) {
  return test;
});`)
});

test('transforms async function expressions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
const xy = async function (test) { return test }`)
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", async function fn(test) {
  return test;
});`)
});

test('transforms named exported function expressions', () => {
  const output = transform(
`/** @firescoutMockFn Config.config */
export const xy = function (test) { return test }`)
  expect(output).toEqual(
`/** @firescoutMockFn Config.config */
export const xy = require("@kaminrunde/cypress-firescout").firescoutMockFn("Config.config", function fn(test) {
  return test;
});`)
});