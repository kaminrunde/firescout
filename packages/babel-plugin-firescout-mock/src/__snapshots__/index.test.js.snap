// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`firescoutMockFn transforms arrow functions with non-block body 1`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", async function fn(test) {
  return 'foo';
});"
`;

exports[`firescoutMockFn transforms async function expressions 1`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", async function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms async functions 1`] = `
"var xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", async function xy(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms declared arrow functions 1`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms declared async arrow functions 1`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", async function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms declared exported arrow functions 1`] = `
"export const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms default exported anonym functions 1`] = `
"export default require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms default exported arrow functions 1`] = `
"export default require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms default exported functions 1`] = `
"export default require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function xy(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms exported functions 1`] = `
"export var xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function xy(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms function expressions 1`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function xy(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms function expressions 2`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms functions 1`] = `
"var xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function xy(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms named exported function expressions 1`] = `
"export const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function fn(test) {
  return test;
});"
`;

exports[`firescoutMockFn transforms named function expressions 1`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockFn(\\"Config.config\\", function xy(test) {
  return test;
});"
`;

exports[`firescoutMockVar tranforms object properties 1`] = `
"const xy = {
  foo: require(\\"@kaminrunde/firescout-utils\\").firescoutMockVar(\\"Config.config\\", 'bar')
};"
`;

exports[`firescoutMockVar transforms declared variables 1`] = `
"const xy = require(\\"@kaminrunde/firescout-utils\\").firescoutMockVar(\\"Config.config\\", {
  foo: 'bar'
});"
`;
